src/main/java/com/jagrosh/jmusicbot/queue/AbstractQueue.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2016 John Grosh (jagrosh).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jagrosh.jmusicbot.queue;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ *
+ * @author Wolfgang Schwendtbauer
+ * @param <T>
+ */
+public abstract class AbstractQueue<T extends Queueable>
+{
+    AbstractQueue(AbstractQueue<T> queue)
+    {
+        this.list = queue != null ? queue.getList() : new LinkedList<>();
+    }
+
+    AbstractQueue()
+    {
+        this.list = new LinkedList<>();
+    }
```

Comment:
I would also remove this constructor here, since we won't be using it anymore. Remove it from `FairQueue` & `LinearQueue`